eva-accession-deprecate/src/main/java/uk/ac/ebi/eva/accession/deprecate/configuration/DeprecableClusteredVariantsReaderConfiguration.java

```diff
@@ -36,7 +37,21 @@
     @StepScope
     DeprecableClusteredVariantsReader deprecableClusteredVariantsReader(MongoClient mongoClient,
                                                                         MongoProperties mongoProperties,
-                                                                        MongoTemplate mongoTemplate) {
-        return new DeprecableClusteredVariantsReader(mongoClient, mongoProperties.getDatabase(), mongoTemplate);
+                                                                        MongoTemplate mongoTemplate,
+                                                                        InputParameters parameters) {
+        boolean assembliesProvided =
+                parameters.getAssemblyAccession() != null && !parameters.getAssemblyAccession().isEmpty();
+
+        if (parameters.getDeprecateAll() == assembliesProvided) {
+            throw new IllegalArgumentException(
+                    "Please provide either: 1) parameters.deprecateAll=true and empty parameters.assemblyAccession or"
+                    + " 2) parameters.deprecateAll=false and parameters.assemblyAccession=<comma-separated-accessions>");
+        }
+        if (assembliesProvided) {
+            return new DeprecableClusteredVariantsReader(mongoClient, mongoProperties.getDatabase(), mongoTemplate,
+                                                         parameters.getAssemblyAccession());
```

Comment:
Should we getAssemblyAccessions instead?

Reply:
see https://github.com/EBIvariation/eva-accession/pull/122#discussion_r269475509. the same explanation applies.